File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.cs
using System;
using System.Text;

public class Solution {
    public string solution(string X, string Y) {
        int[] xArr = new int[10];
        int[] yArr = new int[10];
        StringBuilder sb = new StringBuilder();
        
        foreach(char ch in X) xArr[ch - '0']++;
        foreach(char ch in Y) yArr[ch - '0']++;
        
        for(int i = 9; i >= 0; i--)
        {
            int k = Math.Min(xArr[i], yArr[i]);
            if(k > 0)
            {
                sb.Append(new string((char)(i + '0'), k));
            }
        }
        
        if(sb.Length == 0) return "-1";
        if(sb[0] == '0') return "0";
        return sb.ToString();
    }
}

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string X, string Y) {
    vector<int> cntX(10, 0), cntY(10, 0);
    
    for(char c : X) cntX[c - '0']++;
    for(char c : Y) cntY[c - '0']++;
    
    string answer = "";
    
    for(int i = 9; i >= 0; i--)
    {
        int cnt = min(cntX[i], cntY[i]);
        
        for(int j = 0; j < cnt; j++)
        {
            answer += (i + '0');
        }
    }
    if(answer == "") return "-1";
    if(answer[0] == '0') return "0";
    return answer;
}